### PR TITLE
Adds buildin_getguildonline.

### PR DESCRIPTION
**Pull request by bWolfie**
[//]: # (**********************************)
[//]: # (** Fill in the following fields **)
[//]: # (**********************************)

[//]: # (Note: Lines beginning with syntax such as this one, are comments and will not be visible in your report!)

### Pull Request Prelude

[//]: # (Thank you for working on improving Hercules!)

[//]: # (Please complete these steps and check the following boxes by putting an `x` inside the brackets _before_ filing your Pull Request.)

- [x] I have followed [proper Hercules code styling][code].
- [x] I have read and understood the [contribution guidelines][cont] before making this PR.
- [x] I am aware that this PR will be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed

[//]: # (Describe at length, the changes that this pull request makes.)
Adds BUILDIN(getguildonline) which returns the amount of players online in a guild.
Optional param for type, which will exclude vendor characters from the count.
Returns -1 if guild not found.

I know `getguildmember()` can perform this functionality, but this is quicker and cleaner for scripters IMO.

**Affected Branches:** 

[//]: # (Master? Slave?)
- [x] master
- [x] stable

**Issues addressed:**

[//]: # (Issue Tracker Number if any.)
N/A

### Known Issues and TODO List

[//]: # (Insert checklist here)
[//]: # (Syntax: - [ ] Checkbox)
N/A

[//]: # (**NOTE** Enable the setting "[√] Allow edits from maintainers." when creating your pull request if you have not already enabled it.)

[cont]: https://github.com/HerculesWS/Hercules/blob/master/CONTRIBUTING.md
[code]: https://github.com/HerculesWS/Hercules/wiki/Coding-Style

### CODE REVIEW

**Comment by HerculesWSAPI** [comment]
This change is [Reviewable](https://reviewable.io/reviews/herculesws/hercules/2290)

**Comment by Emistry** [comment]
well, in other way, vending player does consider online too ... it kinda beat the purpose aren't?
`getguildmember` + `isloggedin` could do the same? this could end up become redundant in future.

File: doc/script_commands.txt [hunk 1]
```diff
@@ -3765,6 +3765,15 @@ getarraysize(), because it is not cleared between runs of getguildmember().
 
 For usage examples, see getpartymember().
 
+---------------------------------------
+
+*getguildonline(<guild id>{, <type>});
```
Comment (4144) [inline doc/script_commands.txt]: type better add into constants

File: src/map/script.c [hunk 2]
```diff
@@ -9010,6 +9010,43 @@ static BUILDIN(getguildmember)
...
+
+	if ((g = guild->search(guild_id)) == NULL) {
+		script_pushint(st, -1);
+		return true;
+	}
+	
+	if (script_hasdata(st, 3))
+		type = script_getnum(st, 3);
+
+	if (type == 1) {
```
Comment (4144) [inline src/map/script.c]: and here better use constants too

File: src/map/script.c [hunk 3]
```diff
@@ -9010,6 +9010,43 @@ static BUILDIN(getguildmember)
...
+		struct map_session_data *sd;
+		for (int i = 0; i < MAX_GUILD; i++) {
+			sd = g->member[i].sd;
+			if (sd != NULL && sd->state.vending == 0)
```
Comment (4144) [inline src/map/script.c]: probably also better check ``g->member[i].online``

**Comment by AnnieRuru** [comment]
not really sure about this, but this feels like open a Pandora box
if this is allow, actually mean we are also missing
getclanmemberonline
getpartymemberonline
getpartymemberininstanceonline
getguildmemberininstanceonline
getclanmemberininstanceonline
getinstanceonline
getmaponline --- different from getmapusers because that couldn't count autotraders
getareauseronline -- different from getareausers ....
blah blah blah ....

oh and .... event script included !!
getarrayonline - count all players GID in the array, if player cheat in the event using autotrade

what else ?
I rather stick to getpartymember and getguildmember ... and maybe getunits

ok serious, If I actually want to write this for my private MVP room,
I will also add
`getguildonline(<guild id>{, <type>, <map name>});`
```
if ( !getguildonline( getcharid(CHAR_ID_GUILD), GUILD_ONLINE_ALL, strnpcinfo(NPC_MAP) ) )
mapannounce strnpcinfo(NPC_MAP), "All "+ strcharinfo(PC_GUILD) +" has quit from this event", bc_map;
```

**Pull request merged**